pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -45,14 +45,61 @@ func deviceInfoTask(ctxPtr *zedagentContext, triggerDeviceInfo <-chan struct{})
 
 	for {
 		select {
-		case <-triggerDeviceInfo:
+		case infoType := <-triggerInfo:
+			log.Functionf("reportInfoTask got message for %s", infoType.String())
 			start := time.Now()
-			log.Function("deviceInfoTask got message")
-
-			PublishDeviceInfoToZedCloud(ctxPtr)
-			ctxPtr.iteration++
-			log.Function("deviceInfoTask done with message")
-			ctxPtr.ps.CheckMaxTimeTopic(wdName, "PublishDeviceInfo", start,
+			switch infoType {
+			case info.ZInfoTypes_ZiDevice:
+				PublishDeviceInfoToZedCloud(ctxPtr)
+				ctxPtr.iteration++
+			case info.ZInfoTypes_ZiApp:
+				// publish applications infos
+				sub := ctxPtr.getconfigCtx.subAppInstanceStatus
+				for _, c := range sub.GetAll() {
```

Comment:
This and similar loops in this function can still exceed the 5 minutes. You'd need a trigger per object and not per object type to avoid the watchdog on zedagentdevinfo (the above wdName).

Reply:
Thanks for your comments. I made corrections.